Makefile

```diff
@@ -219,6 +219,21 @@ fmt-check:
 .PHONY: test
 test:
 	GO111MODULE=on $(GO) test -mod=vendor -tags='sqlite sqlite_unlock_notify' $(PACKAGES)
+	@echo "\n==>\033[32m Ok\033[m\n"
+
+PHONY: test-check
+test-check:
+	@echo "Checking if tests have changed the source tree...";
+	@diff=$$(git status -s); \
+	if [ -n "$$diff" ]; then \
+		echo "make test has changed files in the source tree:"; \
+		echo "$${diff}"; \
+		echo "You should change the tests to create these files in a temporary directory."; \
```

Comment:
`/tmp` directory outside gitea source?

Reply:
it doesn't matter where the temporary directory is - it just needs to be deleted at the end of the test.